Frameworks/Foundation/NSThread.mm

```diff
@@ -356,8 +356,15 @@ + (NSThread*)currentThread {
     NSThread* currentThread = [[self class] _threadObjectFromCurrentThread];
 
     if (currentThread == nil) {
-        currentThread = [NSThread new];
+        // Underlying thread created outside the NSThread APIs
+
+        // So create a new instance
+        currentThread = [[NSThread alloc] init];
         [currentThread _associateWithCurrentThread];
```

Comment:
@DHowett-MSFT I don't think non-POD file scope thread_locals are supported by this version of clang (c2?). 

e.g. this is fine at file scope

```
struct Blah {
    Blah() {
        abort();
    }

   ~Blah() {
        abort();
   }
};
 
thread_local Blah blah;
```

But even if thread_local worked the lazy approach avoids the static initialization/destruction overhead for every thread created outside the NS* APIs.

I also haven't confirmed but I think this means there's a leak from thread_local use in NSProgress: https://github.com/Microsoft/WinObjC/blob/develop/Frameworks/Foundation/NSProgress.mm#L50